worker/transcribee_worker/whisper_transcribe.py

```diff
@@ -0,0 +1,179 @@
+import asyncio
+import logging
+from typing import Iterator, Tuple
+
+import requests
+from numpy.typing import ArrayLike
+from transcribee_proto.document import UNKNOWN_SPEAKER, Atom, Paragraph
+from transcribee_worker.config import MODELS_DIR
+from whispercpp import api
+
+
+def get_model_file(model_name: str):
+    whisper_models_dir = MODELS_DIR / "whisper"
+    whisper_models_dir.mkdir(parents=True, exist_ok=True)
+    model_file = whisper_models_dir / f"{model_name}.bin"
+
+    if not model_file.exists():
+        logging.info(f"downloading model {model_name} because it does not exist yet...")
+        base_url = "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main"
+        url = f"{base_url}/ggml-{model_name}.bin"
+        r = requests.get(url, allow_redirects=True)
+        r.raise_for_status()
+        with model_file.open(mode="wb") as f:
+            f.write(r.content)
+
+    return model_file
+
+
+def get_context(model_name: str) -> api.Context:
+    model_file = get_model_file(model_name)
+    logging.info(f"loading model {model_name}...")
+    ctx = api.Context.from_file(str(model_file))
+    ctx.reset_timings()
+    return ctx
+
+
+class TranscriptionWorkDoneToken:
```

Comment:
wouldnt it be more ideomatic if we used `StopIteration` here?

Reply:
I mean not really? This is not a `Iterator`, but I am fine using `StopIteration` if you prefer it anyways.

Reply:
Well actually 
```
ERROR:asyncio:Exception in callback _chain_future.<locals>._set_state(<Future pendi...tasks.py:475]>, <Future at 0x...StopIteration>) at /nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/asyncio/futures.py:379
handle: <Handle _chain_future.<locals>._set_state(<Future pendi...tasks.py:475]>, <Future at 0x...StopIteration>) at /nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/asyncio/futures.py:379>
Traceback (most recent call last):
  File "/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/asyncio/futures.py", line 381, in _set_state
    _copy_future_state(other, future)
  File "/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/asyncio/futures.py", line 357, in _copy_future_state
    dest.set_exception(_convert_future_exc(exception))
TypeError: StopIteration interacts badly with generators and cannot be raised into a Future
```